### ISSUE TITLE
Bad error message on const ref fields

### ISSUE THREAD

**Issue by OmarTawfik** [dotnet/roslyn#17125]
**Valid code:**

```csharp
static void Test(int x)
{
    ref int y = ref x;
}
```
**Invalid code:**

```csharp
static void Test(int x)
{
    const ref int y = ref x;
}
```

**Expected:** an error message on `const` specifying that you cannot use `ref` and `const` modifiers together.

**Found:** parser fails with the following errors:

![image](https://cloud.githubusercontent.com/assets/15987992/22908885/be566078-f205-11e6-9d7c-ce71740ecac7.png)

1. CS1525	Invalid expression term 'ref'
2. CS1002	; expected
3. CS1001	Identifier expected
4. CS8172	Cannot initialize a by-reference variable with a value
5. CS0118	'x' is a variable but is used like a type
6. CS8174	A declaration of a by-reference variable must have an initializer
